c++/CMakeLists.txt

```diff
@@ -51,6 +46,10 @@ if(CAPNP_LITE AND BUILD_TOOLS)
   message(WARNING "Command-line tools will not be built with CAPNP_LITE.")
 endif()
 
+if(BUILD_TOOLS AND EXTERNAL_CAPNP)
+  message(SEND_ERROR "EXTERNAL_CAPNP cannot be set with BUILD_TOOLS.")
```

Comment:
Good point. I had problem to define which tool should be used if both options are set. The answer is obvious now - use external tools.

Question: I would like to remove option BUILD_TOOLS. This option is not supported in autoconf build. 

Question: I would like to rename `EXTERNAL_CAPN`P to `CAPNP_USE_EXTERNAL_TOOLS` and `BUILD_TESTING` to `CAPNP_BUILD_TESTS` to place all cached options into `CAPNP_*` namespace. This PR is backward incompatible anyway.